### PR TITLE
[MAINTENANCE] Mark remaining datasource tests.

### PR DESCRIPTION
**Pull request by billdirks**
- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/datasource/data_connector/test_data_connector.py [hunk 1]
```diff
@@ -52,38 +52,6 @@ def basic_data_connector(tmp_path_factory):
     return basic_data_connector
 
 
-def test_get_batch_definition_list_from_batch_request():
```
Comment (billdirks) [inline tests/datasource/data_connector/test_data_connector.py:55]: This is old code and these aren't tests so I've deleted them.

File: tests/datasource/data_connector/test_runtime_data_connector.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
 
 
 @pytest.fixture
-def basic_datasource_with_assets(tmp_path_factory):
+def basic_datasource_with_assets():
```
Comment (billdirks) [inline tests/datasource/data_connector/test_runtime_data_connector.py:30]: This doesn't actually use the filesystem so updated this fixture and the test markers from filesystem to unit.

Comment (Shinnnyshinshin) [reply tests/datasource/data_connector/test_runtime_data_connector.py:30]: Great change. Thanks

File: tests/datasource/data_connector/test_data_connector.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply tests/datasource/data_connector/test_data_connector.py:55]: 👍🏼

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM!

**Pull request merged**